addon/services/scheduler.js

```diff
@@ -174,11 +174,12 @@ if (DEBUG) {
       if (Ember.testing) {
         this._waiter = () => {
           if (!this.queues) {
-            return;
+            return true;
           }
+
           const lastQueueName = this.queueNames[this.queueNames.length - 1];
           const lastQueue = this.queues[lastQueueName];
-          return !lastQueue.isActive;
+          return !(lastQueue && lastQueue.isActive && !!lastQueue.tasks.length);
```

Comment:
This seems correct, but is a bit hard to reason about. Can we extract any of these to be early returns, or if not assign useful variable names so that the final Boolean logic is more readable?

Reply:
I agree completely, will do.